app/src/main/java/ch/hikemate/app/ui/map/MapScreen.kt

```diff
@@ -45,19 +46,80 @@ import androidx.compose.ui.unit.dp
 import androidx.compose.ui.viewinterop.AndroidView
 import androidx.lifecycle.viewmodel.compose.viewModel
 import ch.hikemate.app.R
+import ch.hikemate.app.model.route.Bounds
+import ch.hikemate.app.model.route.HikeRoute
+import ch.hikemate.app.model.route.LatLong
 import ch.hikemate.app.model.route.ListOfHikeRoutesViewModel
 import org.osmdroid.config.Configuration
 import org.osmdroid.events.MapListener
 import org.osmdroid.events.ScrollEvent
 import org.osmdroid.events.ZoomEvent
 import org.osmdroid.util.GeoPoint
 import org.osmdroid.views.MapView
+import org.osmdroid.views.overlay.Polyline
 
 object MapScreen {
   const val TEST_TAG_MAP = "map"
   const val TEST_TAG_MENU_BUTTON = "menuButton"
   const val TEST_TAG_HIKES_LIST = "hikesList"
   const val TEST_TAG_HIKE_ITEM = "hikeItem"
+
+  const val MIN_HUE = 0
+  const val MAX_HUE = 360
+  const val MIN_SATURATION = 42
+  const val MAX_SATURATION = 98
+  const val MIN_LIGHTNESS = 40
+  const val MAX_LIGHTNESS = 90
+
+  const val STROKE_WIDTH = 10f
+}
+
+/**
+ * Generates a random color in the HSL color space. This function uses the HSL color space to
+ * generate a random color as the HSL is better to generate more visually appealing colors.
+ *
+ * @return The generated color as an [Int].
+ */
+fun getRandomColor(): Int {
+  fun randomInt(min: Int, max: Int): Int {
+    return (min..max).random()
+  }
```

Comment:
Question: is this better than or equivalent to

```kotlin
import kotlin.random.Random

// ...

  fun randomInt(min: Int, max: Int): Int {
    return Random.nextInt(min, max + 1)
  }
```

I mean it does not seem to be a problem so we can keep your version, I think it's more elegant, I was just surprised by the implementation.

Reply:
From my research, both do the same job. From what I understood, your version is a bit more performant as it will not create a range (which is very minimal). The following method would be call in my version:
```kotlin
@SinceKotlin("1.3")
public fun Random.nextInt(range: IntRange): Int = when {
    range.isEmpty() -> throw IllegalArgumentException("Cannot get random in empty range: $range")
    range.last < Int.MAX_VALUE -> nextInt(range.first, range.last + 1)
    range.first > Int.MIN_VALUE -> nextInt(range.first - 1, range.last) + 1
    else -> nextInt()
}
``` 
Which would use your version.

If it's good to you, I will keep my version as it is, as you said, more elegant.